opcotech/elemo
# refactor: revamp UI and related API

## Description

This PR recreates the whole front-end and seriously refactors the API to work better with it. The API will receive more changes when implementing organizations, permissions, and roles.

By taking the chance of recreating the front-end, the new front-end integrates with Storybook as well to allow better visualization of component states.

## Supporting information

- The Storybook interface is available at http://127.0.0.1:6006.
- The oapi-generator is using a [fork](https://github.com/deepmap/oapi-codegen/pull/1102) of the original generator until a feature PR is merged to allow skipping optional field pointers.

### Dependencies

- https://github.com/deepmap/oapi-codegen/pull/1102

### Screenshots

![Screenshot 2023-06-12 at 09-56-20 Elemo](https://github.com/opcotech/elemo/assets/19173947/148626fc-b847-40ce-832e-e0dd8aace03c)
![Screenshot 2023-06-12 at 09-57-03 Elemo](https://github.com/opcotech/elemo/assets/19173947/9db1f503-4a7f-4680-975f-70301ddec439)
![Screenshot 2023-06-12 at 09-57-34 Elemo](https://github.com/opcotech/elemo/assets/19173947/efbc36bd-5a93-483a-bf7f-d429a569788b)
![Screenshot 2023-06-12 at 09-58-15 Gábor Boros Profile Elemo](https://github.com/opcotech/elemo/assets/19173947/23dfdfd8-cd1e-4a18-9e66-23ce7ca504b2)
![Screenshot 2023-06-12 at 09-58-25 Settings Elemo](https://github.com/opcotech/elemo/assets/19173947/4580100c-5d31-46e8-90c0-13b937792ca3)
![Screenshot 2023-06-12 at 09-58-32 Security settings Elemo](https://github.com/opcotech/elemo/assets/19173947/95a6c0a0-c064-45aa-868c-6e2e87e022cf)
![Screenshot 2023-06-12 at 09-58-42 System settings Elemo](https://github.com/opcotech/elemo/assets/19173947/7b85a69a-c618-4048-b042-6f6cc0ed0ced)
![Screenshot 2023-06-12 at 09-58-53 Elemo](https://github.com/opcotech/elemo/assets/19173947/dab5be33-aecc-4e04-9393-a649a67f1fda)

## Testing instructions

- Login using wrong credentials
- Login with correct credentials
- Click around
- Try to update settings
- Check profile
- Check license
- Login with non privileged user and check system settings

## Other information

N/A

## Checklist

### Documentation

- [ ] Documentation comments are updated
- [ ] Documentation site is updated

### Code quality

- [ ] `make format` has been run

### Review

- [ ] Self-checked the diff
- [ ] Reviewed the code based on Go's [code review guide]
- [ ] Reviewed the code based on Go's [concurrency review guide]

### Miscellaneous



[code review guide]: https://github.com/golang/go/wiki/CodeReviewComments
[concurrency review guide]: https://github.com/golang/go/wiki/CodeReviewConcurrency
